### ISSUE TITLE
TacOps Active Probe applying -1 to-hit against unit in TacOps Woods Cover

### ISSUE THREAD

**Issue by Saklad5** [MegaMek/megamek#4927]
### Environment
MegaMek Version: MekHQ v0.49.15 Development Snapshot
Operating System: Mac
Java Version: 20.0.1

### Description

When using the optional rule for Active Probes, MegaMek currently applies a flat -1 to-hit modifier when an attacker with such equipment fires at a target in or behind woods cover. This normally negates the +1 modifier applied by the woods hex. In the event that the optional Woods Cover rule is also being used, however, that +1 modifier no longer exists without intervening woods hexes.

While the rule as currently written supports MegaMek's behavior, I've gotten an explicit [ruling](https://bg.battletech.com/forums/index.php/topic,82931.0.html) on the forum that active probes should not apply a benefit unless a to-hit penalty from woods is actually in effect.

In other words, if TacOps Woods Cover is being used, TacOps Active Probes should only apply a to-hit bonus if _intervening_ woods hexes exist.
